wagtail/admin/templatetags/util.py

```diff
@@ -0,0 +1,124 @@
+from inspect import getfullargspec, unwrap
+
+from django.core.exceptions import ImproperlyConfigured
+from django.template.library import InclusionNode, parse_bits
+
+
+class PartialNode(InclusionNode):
+    """
+    Splits an existing template into to based on a string provided in the template
+    to act as two separate virtual (partial) templates for a start & end either
+    side of the split string.
+    The split string is not included in either final template.
+    """
+
+    def __init__(self, *args, **kwargs):
+        is_end = kwargs.pop("is_end", False)
+        split_tag = kwargs.pop("split_tag")  # required
+
+        super().__init__(*args, **kwargs)
+
+        self.is_end = is_end
+        self.split_tag = split_tag
+
+    def render(self, context):
+        template = context.template.engine.get_template(self.filename)
+        source = template.source
+        parts = source.split(self.split_tag)
+
+        # patch in a template that is just a partial version of the original & re-apply metadata for debugging
+        self.filename = context.template.engine.from_string(
+            parts[1] if self.is_end else parts[0]
+        )
+        self.filename.name = template.name
+        self.filename.origin = template.origin
+        self.filename.source = template.source
+
+        content = super().render(context)
+
+        return content
+
+
+class SplitTemplateTag:
+    """
+    Base class for building a split template tag in the format.
+    `{% birdbath %} {% birdbathend %}`
```

Comment:
The header tag you've used in[wagtail/admin/templates/wagtailadmin/home.html](https://github.com/wagtail/wagtail/pull/8775/files#diff-1044caae0d6e7ac62c485c45557c5f9afd961bce3b05efcd9b5c503156159f3b) above doesn't follow this format - it has `{% endheader %}` rather than `{% headerend %}`